### PR TITLE
Migrate to null-safety

### PR DESCRIPTION
**Pull request by ipcjs**
#11

### CODE REVIEW

**Comment by kostaa** [comment]
Hi there. Is there an issue with this pull request? If not could you merge and release please?
This is a blocker for several libraries (reselect, re_reselect) and many apps that use Redux.

Many thanks!

**Comment by kostaa** [comment]
Hi @brianegan, 
Looks like this repo has been abandoned. 
Would you be willing to fork it into your repo and upgrade to null-safety (and reselect_dart)?

**Comment by sinegovsky-ivan** [comment]
@kostaa we will review and merge this MR. ETA - 1 week. 
sorry for the delay.

**Comment by kostaa** [comment]
Great! In my case there are another two libraries that depend on this one before I can fully switch my app to null-safety. 
Would you be able to do it maybe a bit earlier? 
Many thanks.

File: pubspec.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: memoize
-version: 2.0.0
+version: 3.0.0
```
Comment (ipcjs) [inline pubspec.yaml:2]: This update is forward compatible, maybe we shouldn't update the major version number?

**Comment by missvalentinep** [comment]
@ipcjs hey! Great merge request, sorry for the delay. 

Could you please:
1. Update the changelog
2. Perhaps, write something about supporting null-safety to readme?

File: pubspec.yaml (hunk 1 above)
Comment (missvalentinep) [reply pubspec.yaml:2]: No, let's keep this a major change.

**Pull request merged**